### PR TITLE
IGAPP-333: Documentation

### PR DESCRIPTION
**Pull request by klinzo**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

**Comment by klinzo** [comment]
Build configs will be merged and moved in https://issues.tuerantuer.org/browse/IGAPP-271.

**Review by Taggotty — APPROVED** [review]

File: web/README.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+* Run > Edit Configurations > Defaults > Jest:
+    * Set *Configuration file* to *jest.config.json*
+* Settings > Languages & Frameworks > JavaScript and
+    * Choose *Flow* as Language version
+    * Set *Flow package or executable* to *<project_dir>/node_modules/flow-bin*
+* [optional] Associate the *\*.snap* files with the file type *JavaScript*.
+* [optional] Install the following plugins:
+    * [Styled Components](https://plugins.jetbrains.com/plugin/9997-styled-components--styled-jsx/)
+    * [EJS](https://plugins.jetbrains.com/plugin/index?xmlId=com.jetbrains.lang.ejs)
```
[the remark this answers is not in the document]
Comment (klinzo) [reply web/README.md:38]: Worked for me sometime ago at least. Will check

**Comment by klinzo** [comment]
I am quite unsure as to what belongs in the common documentation and what belongs just in the separate projects. If you think anything is wrong in this concern, let me know! (@Taggotty ping because you already reviewed)

File: native/docs/windows-setup.md [hunk 2]
```diff
@@ -1,6 +1,8 @@
 # Windows Setup
 
-Since we noticed some issues with the usage of windows which can be happened, here a small setup guide
+Since we noticed some issues with the usage of windows which can be happened, here a small setup guide.
+
+*Note: This guide may be outdated. You may want to use the [Windows Subsystem for Linux](../../docs/wsl-setup.md) instead.*
```
[the remark this answers is not in the document]
Comment (klinzo) [reply native/docs/windows-setup.md:5]: Do you mean the wsl docu is mostly targeted at the web? I think that is fine.

Yeah maybe in the future :D I think we should not waste any resources on this windows topic. We don't have a windows user anymore and we strongly suggest to use linux or macos now... :grin:

File: README.md [hunk 3]
```diff
@@ -2,88 +2,65 @@
...
-
-* Own a Mac or another Apple device.
-* Install [XCode](https://developer.apple.com/xcode/).
-
-*Note: In order to work with the XCode project, always open `ios/Integreat.xcworkspace`.*
+* Install [yarn](https://yarnpkg.com/).
 
 ### Project setup
+
 We suggest **[IntelliJ IDEA Ultimate](https://www.jetbrains.com/idea/)** as IDE. JetBrains provides free licences for students.
```
Comment (Taggotty) [inline README.md:52]: Maybe mention what can be used as an alternative. Maybe @Znippet has some input to this :)

File: README.md [hunk 4]
```diff
@@ -2,88 +2,65 @@
...
 If you are using a different IDE, the steps might differ.
 
 * Import this project (VCS > Get from Version Control).
-* Run `yarn` in the terminal to install all dependencies.
```
Comment (Taggotty) [inline README.md:49]: If you remove this part it needs to be mentioned that the complete setup differs depends on the projet, e.g. native or web or sth like that.

**Review by Taggotty — APPROVED** [review]
I've also worked on the documentation a week ago and found it also very difficult to find the line where to split the specific documentation and where not. I feel like there is no right or wrong here though. Better repeat yourself a little is what I prefer.

File: README.md (hunk 3 above)
Comment (klinzo) [reply README.md:52]: I don't think this necessary to be honest :D I mean, every IDE can be used... If one wants, even a simple text editor is possible :smile:

File: README.md (hunk 4 above)
Comment (klinzo) [reply README.md:49]: Is the following not enough? `* Follow the steps mentioned in the projects README.`

File: native/docs/windows-setup.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (Taggotty) [reply native/docs/windows-setup.md:5]: > Do you mean the wsl docu is mostly targeted at the web? I think that is fine.
Right.

File: README.md (hunk 3 above)
Comment (Taggotty) [reply README.md:52]: sure. I was just thinking of those poor people out there who cannot scrounge an Ultimate Liscence. But sure fair enough you've got a point. 👍🏼

File: README.md (hunk 4 above)
Comment (Taggotty) [reply README.md:49]: Sure

Comment (klinzo) [reply README.md:49]: What do you mean by sure? :D If you tell me what to write I'll include it :sunglasses:

Comment (Taggotty) [reply README.md:49]: Oh wow. Somehow I kind of missed that. Sorry. You did well 👍🏼

File: web/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (klinzo) [reply web/README.md:38]: Works for me

**Review by maxammann — APPROVED** [review]

**Pull request merged**